### PR TITLE
Feature/cmp 878/refactor backend config: Refactor backend configuration in product helm chart

### PR DESCRIPTION
**Pull request by saudkhan116**
# Why we create this PR?
 
Refactor backend configuration in the product helm chart `templates/backend-config.yaml` and values files to disable/hide the app core configuration from the user and only allow user to configure user-related settings as default.
 
# What we want to achieve with this PR?
 
- To have a reference implementation, the backend needs to be configured directly through the `values.yaml` files and not as a fixed yaml payload
- To enable the specific backend configuration from the helm values file to follow good coding practices
 
# What is new?
 
## Updated

| Tickets |
| :---:   |
| [cmp-878](https://jira.catena-x.net/browse/CMP-878) |

### CODE REVIEW

File: charts/digital-product-pass/values-int.yaml [hunk 1]
```diff
@@ -77,128 +77,57 @@ backend:
...
+    xApiKey: <path:material-pass/data/int/edc/oauth#api.key>
+    participantId:  <path:material-pass/data/int/edc/participant#bpnNumber>
+    endpoint: "https://materialpass.int.demo.catena-x.net/consumer"
+    management: "/management/v2"
+    catalog: "/catalog/request"
+    negotiation: "/contractnegotiations"
+    transfer: "/transferprocesses"
+    delay: 100
 
+  url: "materialpass.int.demo.catena-x.net"
```
Comment (matbmoser) [inline charts/digital-product-pass/values-int.yaml:92]: We needed to get this URL from the ingress configuration

File: charts/digital-product-pass/templates/configmap-backend.yaml [hunk 2]
```diff
@@ -26,5 +26,122 @@ metadata:
...
+        tokenUri: {{ tpl (.Values.backend.configuration.keycloak.tokenUri | default "http://localhost") . | quote }}
+        userInfoUri: {{ tpl (.Values.backend.configuration.keycloak.userInfoUri | default "http://localhost") . | quote }}
+      # -- edc consumer connection configuration
+      edc:
+        endpoint: {{ .Values.backend.edc.endpoint }}
+        management: {{ .Values.backend.edc.management }}
+        catalog: {{ .Values.backend.edc.catalog }}
+        negotiation: {{ .Values.backend.edc.negotiation }}
+        transfer: {{ .Values.backend.edc.transfer }}
+        receiverEndpoint: "https://{{ .Values.backend.url }}/endpoint"
```
Comment (matbmoser) [inline charts/digital-product-pass/templates/configmap-backend.yaml:62]: Backend url should come from the Ingress to auto configure and not have data redundancy

**Review by matbmoser — APPROVED** [review]

File: charts/digital-product-pass/templates/configmap-backend.yaml (hunk 2 above)
Comment (saudkhan116) [reply charts/digital-product-pass/templates/configmap-backend.yaml:62]: We used a variable "hostname" to populate the backend url from the ingress configuration

File: charts/digital-product-pass/values-int.yaml (hunk 1 above)
Comment (saudkhan116) [reply charts/digital-product-pass/values-int.yaml]: The url now contains the value from ingress

File: charts/digital-product-pass/values-beta.yaml [hunk 3]
```diff
@@ -69,134 +69,67 @@ backend:
...
           - path: /
             pathType: Prefix
     tls:
       - secretName: tls-secret
         hosts:
-          - materialpass.beta.demo.catena-x.net
+          - *hostname
+
+  edc:
+    clientId: <path:material-pass/data/beta/backend#clientId>
```
Comment (matbmoser) [inline charts/digital-product-pass/values-beta.yaml:82]: I think the Client ID and the Client Secret could be included in a "oauth" key better then the "edc" configuration
actualy we need them just for the BPN Discovery, EDC Discovery, IRS and not the EDC.

File: charts/digital-product-pass/values-dev.yaml [hunk 4]
```diff
@@ -69,134 +69,66 @@ backend:
...
+          - *hostname
 
-  avp:
-    helm:
-      clientId: <path:material-pass/data/dev/backend#clientId>
-      clientSecret: <path:material-pass/data/dev/backend#clientSecret>
-      xApiKey: <path:material-pass/data/dev/edc/oauth#api.key>
-      participantId:  <path:material-pass/data/dev/edc/participant#bpnNumber>
+  edc:
+    clientId: <path:material-pass/data/dev/backend#clientId>
```
Comment (matbmoser) [inline charts/digital-product-pass/values-dev.yaml:82]: same here

File: charts/digital-product-pass/values-int.yaml [hunk 5]
```diff
@@ -68,137 +68,66 @@ backend:
...
-          include-binding-errors: ALWAYS
-          include-stacktrace: ON_PARAM
-          include-exception: false
-        port: 8888
-        tomcat:
-          max-connections: 10000
+          - *hostname
 
+  edc:
+    clientId: <path:material-pass/data/int/backend#clientId>
```
Comment (matbmoser) [inline charts/digital-product-pass/values-int.yaml:81]: same here

**Review by matbmoser — CHANGES_REQUESTED** [review]

File: charts/digital-product-pass/values-dev.yaml [hunk 6]
```diff
@@ -69,134 +69,67 @@ backend:
...
+  url: *hostname
+  serverPort: 8888
+  logging:
+    level:
+      root: INFO
+      utils: INFO
+  configuration:
+    maxRetries: 10
+    keycloak:
+      realm: CX-Central
```
Comment (matbmoser) [inline charts/digital-product-pass/values-dev.yaml:103]: Shall We also include this configuration there?

File: charts/digital-product-pass/values.yaml [hunk 7]
```diff
@@ -93,143 +93,95 @@ backend:
...
+  # -- passport data transfer configuration
+  passport:
+    # -- passport versions and aspects allowed
+    aspects:
+      - "urn:bamm:io.catenax.generic.digital_product_passport:1.0.0#DigitalProductPassport"
+      - "urn:bamm:io.catenax.battery.battery_pass:3.0.1#BatteryPass"
+      - "urn:bamm:io.catenax.transmission.transmission_pass:1.0.0#TransmissionPass"
+  digitalTwinRegistry:
+    # -- central digital twin registry url
+    url: ""
```
Comment (matbmoser) [inline charts/digital-product-pass/values.yaml:159]: We forgot to remove this

**Review by matbmoser — CHANGES_REQUESTED** [review]

File: charts/digital-product-pass/values.yaml (hunk 7 above)
Comment (matbmoser) [reply charts/digital-product-pass/values.yaml:159]: I will do it

Comment (saudkhan116) [reply charts/digital-product-pass/values.yaml:159]: Please do it by yourself whatever you would like to have, thanks 👍

**Review by matbmoser — APPROVED** [review]

**Comment by matbmoser** [comment]
Ok now it is there and ready. The charts are also updated to the latest version.

**Pull request merged**